Src/FluentAssertions/Types/TypeSelector.cs

```diff
@@ -293,12 +293,8 @@ public TypeSelector UnwrapTaskTypes()
     {
         types = types.ConvertAll(type =>
         {
-            if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Task<>))
-            {
-                return type.GetGenericArguments().Single();
-            }
-
-            if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(ValueTask<>))
+            if (type.IsGenericType && type.GetGenericTypeDefinition() is var genericTypeDefinition
```

Comment:
🤔 I'm on the brink whether I actually like this.

Reply:
Care to share what you don't like :wink: 

The thing about the current code that annoys me a bit is that we check `type.IsGenericType` twice.
It's like the code yells at me "I JUST TOLD YOU THAT I'M A GENERIC TYPE! :rage:"

How do you feel about this less condensed version then?
```cs
if (type.IsGenericType)
{
    var genericTypeDefinition = type.GetGenericTypeDefinition();
    if (genericTypeDefinition == typeof(Task<>) || genericTypeDefinition == typeof(ValueTask<>)))
    {
        return type.GetGenericArguments().Single();
    }
}
```

Reply:
That would work well. I think my main issue was with the nested _ands_ and _ors_.